Draft of design outline

AutumnVulpes opened the pull request.

AutumnVulpes committed:
Draft of design outline

AutumnVulpes committed:
Update of design doc

Added fields required for SFC and consideration of some issues

methylDragon commented:
Could you provide a description

methylDragon reviewed (changes requested):
This format is hard to parse, I don't understand what the bullets are for since there are no sections and no descriptions of those sections.

Please use the appropriate headers, making sure to cover:

- What are you trying to do?
- Why are you doing it?
- Why is the method you picked better than the other alternatives (and what are those alternatives?)

AutumnVulpes committed:
Csv formatter design doc

Rewriting and polishing of the design doc, including background, context/use case, and proposed design. Alternatives still WIP.

AutumnVulpes commented:
Updated design doc, reverted back to draft.
Completed: Introduction, Context and Design flow.
WIP: Alternatives considered
